### PR TITLE
Refactored ImportBookmarkLinks using newly available tools from Java 1.8

### PR DESCRIPTION
**Pull request by mortenbruhn**
## Description
- Refactored `ImportBookmarkLinks` using some of the tools made available by #14181
- Refactored `ImportBookmarkLinksTest` to use `Parameterized`
- Moved `UrlToIdParser` and implementing classes to separate files to allow for re-use elsewhere

## Related issues
- #14777
- #14181

### CODE REVIEW

**Comment by bekuno** [comment]
Ok to test

**Review by eddiemuc — COMMENTED** [review]
Code is fine. It is quite extensive for the problem at hand, but I know from own experience that Java is just very noisy :-) 

I hesitate from merging for now because I am not sure whether we will leave in desugaring for good. In case we have to remove it again, then this PR has to be reverted - probably manually. 
@mortenbruhn would you be willing to do such a revert in this case?

**Comment by mortenbruhn** [comment]
> Code is fine. It is quite extensive for the problem at hand, but I know from own experience that Java is just very noisy :-)
> 
> I hesitate from merging for now because I am not sure whether we will leave in desugaring for good. In case we have to remove it again, then this PR has to be reverted - probably manually. @mortenbruhn would you be willing to do such a revert in this case?

AFAIK, desugaring is only necessary in this case if the Android API version in `minSdk` remains at level 21. If we upgrade to a more recent one, the need for desugaring goes away.

TL;DR: Yes, I'll take my chances :)

**Comment by eddiemuc** [comment]
> AFAIK, desugaring is only necessary in this case if the Android API version in `minSdk` remains at level 21. If we upgrade to a more recent one, the need for desugaring goes away.

That's true. This has been discussed before, and the team wishes to keep support for SDK 21 alive. So unfortunately upgrade is not an option.

> TL;DR: Yes, I'll take my chances :)

All right, let's take our chances together :) Thanks a lot for your contribution!
